src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -507,23 +487,16 @@ else if (exercise instanceof ProgrammingExercise) {
         course.setExercises(interestingExercises);
 
         for (Exercise exercise : interestingExercises) {
-            long numberOfSubmissions = 0L;
-            if (exercise instanceof TextExercise) {
-                numberOfSubmissions = textSubmissionService.countSubmissionsToAssessByExerciseId(exercise.getId());
+            long numberOfSubmissions;
+            if (exercise instanceof ProgrammingExercise) {
+                numberOfSubmissions = programmingExerciseService.countSubmissionsByExerciseIdSubmitted(exercise.getId());
             }
-            else if (exercise instanceof ModelingExercise) {
-                numberOfSubmissions += modelingSubmissionService.countSubmissionsToAssessByExerciseId(exercise.getId());
+            else {
+                numberOfSubmissions = submissionRepository.countByExerciseIdSubmittedBeforeDueDate(exercise.getId());
```

Comment:
Couldn't this method also be used for programming exercises? Since we are quering a specific exercise, the WHERE condition in the SQL feels a little unnecessary.

Reply:
Not really. The programming exercise needs a special handling to not be counted twice